lib/internal/modules/esm/resolve.js

```diff
@@ -135,19 +134,12 @@ function emitLegacyIndexDeprecation(url, packageJSONUrl, base, main) {
 
 const realpathCache = new SafeMap();
 
-/**
- * @param {string | URL} path
- * @returns {import('fs').Stats}
- */
-const tryStatSync =
-  (path) => statSync(path, { throwIfNoEntry: false }) ?? new Stats();
-
 /**
  * @param {string | URL} url
  * @returns {boolean}
  */
 function fileExists(url) {
-  return statSync(url, { throwIfNoEntry: false })?.isFile() ?? false;
+  return internalModuleStat(url.pathname ?? url) === 0;
```

Comment:
If url is an `URL`, we must call `fileURLToPath` to handle special cases.

Reply:
Thanks for the review @targos. Even better: I replaced this with `toPathIfFileURL`